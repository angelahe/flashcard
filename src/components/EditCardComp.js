import React from 'react';
import PropTypes from 'prop-types';
import AddImageComp from './AddImageComp';
import '../styles/flashcard.css';
import Client from './Client';
import done from '../img/buttons/done_FFFFFF.png';
import english from '../img/buttons/english_FFFFFF.png';
import spanish from '../img/buttons/spanish_FFFFFF.png';
import order from '../img/buttons/order_FFFFFF.png';

class EditCardComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentCard: '',
      currentL1: '',
      currentL2: '',
      currentOrder: 0,
      currentImageUrl: 'https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png',
      currentImageId: 77680,
      addImage: false,
      styleImage: 'cardContainer',
    };
  }

  componentDidMount = () => {
    const { card } = this.props;

    this.setState({
      currentL1: card.L1_word,
      currentL2: card.L2_word,
      currentImageUrl: card.img_url,
      currentImageId: card.img_id,
      currentOrder: card.card_order,
    });
  }

  handleL1Change = (event) => {
    this.setState({ currentL1: event.target.value });
  };

  handleL2Change = (event) => {
    this.setState({ currentL2: event.target.value });
  };

  handleOrderChange = (event) => {
    this.setState({ currentOrder: event.target.value });
  };

  handleEditCardClick = () => {
    const { card, onCardEdited } = this.props;
    const {
      currentL1,
      currentL2,
      currentOrder,
      currentImageUrl,
      currentImageId,
    } = this.state;

    Client.editCard(card.card_id, currentL1, currentL2, currentOrder,
      currentImageUrl, currentImageId)
      .then((id) => {
        onCardEdited(id);
      });
  };

  handleImageAdded = (imageUrl, imageId) => {
    console.log(`image url: ${imageUrl} image id: ${imageId}`);
    this.setState({
      currentImageUrl: imageUrl,
      currentImageId: imageId,
      addImage: false,
      styleImage: 'cardContainer',
    });
  }

  handleImageClick = () => {
    this.setState({ addImage: true });
  }

  render() {
    const {
      currentL1,
      currentL2,
      currentOrder,
      currentImageId,
      currentImageUrl,
      addImage,
      styleImage,
    } = this.state;
    return (
      <div className="EditCardComp">
        { (addImage !== true)
          ? (
            <div className="AddImage">
              <div className="LineContainer">
                <button
                  className={styleImage}
                  type="button"
                  onClick={this.handleImageClick}
                  onKeyPress={this.handleItemClick}
                >
                  <img
                    className="cardImg"
                    src={currentImageUrl}
                    alt={currentImageId}
                  />
                </button>
              </div>
              <br />
              <div>
                <div className="LineContainer">
                  <img className="btnImg" src={english} alt="English" />
                  <input className="L1Input DetailText" value={currentL1} onChange={this.handleL1Change} />
                  <br />
                </div>
                <div className="LineContainer">
                  <img className="btnImg" src={spanish} alt="Spanish" />
                  <input className="L2Input DetailText" value={currentL2} onChange={this.handleL2Change} />
                </div>
                <div className="LineContainer">
                  <img className="btnImg" src={order} alt="Order" />
                  <input className="Order DetailText" type="Number" value={currentOrder} onChange={this.handleOrderChange} />
                </div>
                <div className="LineContainer">
                  <button className="DoneBtn AppBtn" type="button" onClick={this.handleEditCardClick}>
                    <img className="btnImg" src={done} alt="Done" />
                  </button>
                </div>
              </div>
              <br />
              <span className="DetailText">Image:</span>
              <p>Url:{currentImageUrl}  id:{currentImageId}</p>
              <img className="btnImg" src={currentImageUrl} alt={currentImageId} />
            </div>
          )
          : <AddImageComp onImageAdded={this.handleImageAdded} />
        }
      </div>
    );
  }
}

EditCardComp.defaultProps = {
  onCardEdited: () => { },
  card: {
    L1_word: '',
    L2_word: '',
    img_id: 0,
    img_url: '',
    card_order: 0,
    cardtype_id: '',
  },
};

EditCardComp.propTypes = {
  card: PropTypes.shape({
    card_id: PropTypes.string.isRequired,
    deck_id: PropTypes.string.isRequired,
    L1_word: PropTypes.string,
    L2_word: PropTypes.string,
    img_id: PropTypes.number,
    img_url: PropTypes.string,
    card_order: PropTypes.number,
    cardtype_id: PropTypes.string,
  }),
  onCardEdited: PropTypes.func,
};

export default EditCardComp;
