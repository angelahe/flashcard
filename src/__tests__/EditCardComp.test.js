import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import EditCardComp from '../components/EditCardComp';

const card = 'ABCD-1234';
it('renders without crashing', () => {
  shallow(<EditCardComp card={card} />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EditCardComp card={card} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<EditCardComp card={card} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div EditCardComp on shallow render', () => {
  const component = shallow(<EditCardComp card={card} />);
  expect(component.html()).toContain('EditCardComp');
});
