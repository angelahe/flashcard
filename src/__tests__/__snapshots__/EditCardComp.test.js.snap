// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="EditCardComp"
>
  <div>
    <button
      className="cardDefault"
      onClick={[Function]}
      type="button"
    >
      <img
        alt={77680}
        className="cardImg"
        src="https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png"
      />
    </button>
    <br />
    <div>
      <div
        className="LineContainer"
      >
        <img
          alt="English"
          className="btnImg"
          src="english_FFFFFF.png"
        />
        <input
          className="DetailText"
          onChange={[Function]}
          value=""
        />
        <br />
      </div>
      <div
        className="LineContainer"
      >
        <img
          alt="Spanish"
          className="btnImg"
          src="spanish_FFFFFF.png"
        />
        <input
          className="DetailText"
          onChange={[Function]}
          value=""
        />
      </div>
      <button
        className="AppBtn"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Done"
          className="btnImg"
          src="done_FFFFFF.png"
        />
      </button>
    </div>
    <br />
    <span
      className="DetailText"
    >
      Image:
    </span>
    <p>
      Url:
      https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png
        id:
      77680
    </p>
    <img
      alt={77680}
      className="btnImg"
      src="https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png"
    />
  </div>
</div>
`;
